### ISSUE TITLE
Unreachable code at SignedInteger.java:[line 102]

### ISSUE THREAD

**Issue by hallo02** [empeeoh/BACnet4J#2]
else if (smallValue < Short.MAX_VALUE && smallValue > Short.MAX_VALUE) {
                length = 2;
}

should be

else if (smallValue < Short.MAX_VALUE && smallValue > Short.MIN_VALUE) {
                length = 2;
}

isn't it?

**Comment by mlohbihler** [comment]
Yes, it should. Thanks. I've made this change in v4. (Not yet committed at this moment.)
